src/KubernetesClient/Cache/Store.cs

```
@@ -0,0 +1,88 @@
+using System;
+using k8s.Models;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace k8s.cache
+{
+    public class Store : IStore
+    {
+        private IThreadSafeStore _threadSafeStorage;
+        private Func<IKubernetesObject, string> _metaNamespaceFunc;
+        public string MetaNamespaceKeyFunc(IKubernetesObject o) 
+        {
+            try {
+                 var typedObject = (V1ObjectMeta)o.GetType().GetProperty("Metadata").GetValue(o, null);  
+                if (typedObject.NamespaceProperty == "") {
+                    return typedObject.Name;                
+                }
+                return typedObject.NamespaceProperty+"/"+typedObject.Name;
+            } catch (Exception ex) {
+                Console.WriteLine(ex);
```

Do we really want to swallow the exception here?